_posts/2015-07-20-third-party-integration.md

```diff
@@ -0,0 +1,85 @@
+In every of these areas new startup companies are founded that offer new services.
+At the same time, also existing companies expand their service portfolio in order not to lose the connection in the dynamic e-commerce market.
+These circumstances keep us busy not only in integrating periodically new services but also in maintaining the existing integrations.
+
+If companies expand abroad, as it is the case at ePages, it might be required to include additional services, to being attractive for the merchants in that specific country.
+We integrated USPS and FedEx, since it would not make any sense to enter the US market without an integration of the most important US shipping agents.
+
+# Third-party libraries vs Third-party web services
+
+Today's software is mostly based on third-party components, as there are already proven solutions in place that can be easily reused.
+Furthermore, these project are supported by great communities that continuously develop the software and often are absolute experts in this respective field.
+
+However, with regards to third-party components we have to distinguish between third-party library and third-party web service.
+Third-party libraries are integrated directly into the code base and remain unchanged until consciously a new library version will be included.
+
+Working with third-party web services is more complex.
+Even if a third-party web service, other than a third-party library, is not directly expands the code base, one must never forget:
+By integrating a third-party service, this service will be part of the own software.
+The software will be expanded by functionality that is dependent on the availability of the third-party web service and acts as expected.
+This fact puts special demands on the developers.
+
+# The challenge
+
+Mostly one knows only the endpoint of the third-party API and only a more or less useful documentation is available.
+The problem is, that not only the web service itself, but also both, the endpoint and the documentation, can change every time.
+You will only notice the change, if something does not work anymore.
+Sometimes an SDK exists for the API that should facilitate working with it.
+But sometimes the SDK does not support the latest API version or the required programming language is not available.
+
+Often the documentation was only produced in the national language of the API developers making it impossible to understand the content.
+The result is that integration problems cannot be fixed by our developers in Barcelona if the documentation is e.g. only available in German.
+
+# The approach
+
+Ideally, we are in direct contact with the developers of the respective third-party to clarify any questions or concerns.
+Unfortunately, that is not always the case and we have to find it out by ourselves with the trial-and-error method or reverse engineering.
```

Comment:
"Unfortunately, that is not always the case and we have to find it out by ourselves with the trial-and-error method or reverse engineering." Warum ist das so?